test/util/nightwatch-commands/printLogs.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Outputs to the console all the console.log()s called from the browser since
+ *  the last client.printLogs() was called.
+ *
+ * NOTE: This only works from log level WARNING and above right now. This has to
+ *  do with the selenium log level, methinks, and I'm not sure how to change that.
+ *  To "log" from within the browser, use console.warn() for now.
+ *  Also, this doesn't work with console.warn('foo:', foo). To output a variable,
+ *  be sure to include it in a string template: console.warn(`foo: ${JSON.stringify(foo)}`)
+ *  It's obnoxious, but that's the only way I've found that works well.
+ *
+ * @param {String} level     The level of logs to print. Note: This can cause
+ *                           some logs to be skipped. For all log levels, See
+ *                           https://github.com/SeleniumHQ/selenium/wiki/Logging#log-levels
+ *                           If no level is provided, all levels are output.
+ * @param {bool}   printAll  Prints all the logs from 0 on. In case some logs
+ *                           got skipped because of log level specificity.
+ */
+exports.command = function printLogs(level, printAll = false) {
```

Comment:
It's not great, but it opens up some logging from inside our app (with some caveats, as mentioned in the comments). To use this, just call `client.printLogs()` after you know the app logged some things. Of course, if the test fails before you can call this, it's significantly less than helpful, but I couldn't find a way to point the browser's `console.log` to the `console.log` of the running command, so it's the best I could do.